test/com/minseungkwon/datastructure/Graph/BinaryTreeTest.java

```diff
@@ -319,4 +322,20 @@ public void test_findCommonAncestorUsingPostOrder() {
         assertEquals(7, bt.findCommonAncestorUsingPostOrder(7, 9).data);
         assertEquals(null, bt.findCommonAncestorUsingPostOrder(7, 11));
     }
+
+    @Test
+    public void test_findAllCombination(){
+        Integer[] arr = makeArrayForTree(5);
+        BinaryTree<Integer> bt = new BinaryTree<>();
+        bt.makeTree(arr);
+        ArrayList<LinkedList<Node>> result = bt.findAllCombination(bt.getRoot());
+        StringBuilder s = new StringBuilder();
+        for(LinkedList<Node> ll : result){
+            for(Node n : ll){
+                s.append(n.data);
+            }
+            s.append(" ");
+        }
+        assertEquals("20134 20314 20341 23014 23041 23401 ", s.toString());
```

Comment:
I might miss the details but why does it start with 2?

Reply:
Since when it makes a tree with inputs {0,1,2,3,4}, the mid number becomes the root. In this case, number 2 is the mid number, so it became the root. That's why it starts with number 2.